torch/nn/modules/transformer.py

```diff
@@ -195,6 +214,7 @@ def __init__(self, encoder_layer, num_layers, norm=None, enable_nested_tensor=Tr
         # this attribute controls whether nested tensors are used
         self.use_nested_tensor = enable_nested_tensor
         self.mask_check = mask_check
+        self.seq_len_pos = 1 if encoder_layer.self_attn.batch_first else 0
```

Comment:
I see from the docs that that sometimes `src` can be unbatched of shape `(S, E)`. I am not sure what the value of `batch_first` will be then, just checking whether this accounts for the unbatched case

Reply:
Good catch!!!! In that case it would be 0, right?  So we have to check the dimension of the input data, and hardcode position 0 if the input is 2 dimensional.

Reply:
OK, I added _get_seq_len() function that handles all scenarios - nested tensors, unbatched tensors and batched tensors with batch first = {True, False}.